osu.Game/Overlays/BeatmapSet/LeaderboardModSelector.cs

```diff
@@ -54,7 +54,7 @@ private void onRulesetChanged(ValueChangedEvent<RulesetInfo> ruleset)
                 return;
 
             modsContainer.Add(new ModButton(new ModNoMod()));
-            modsContainer.AddRange(ruleset.NewValue.CreateInstance().GetAllMods().Where(m => m.UserPlayable).Select(m => new ModButton(m)));
+            modsContainer.AddRange(ruleset.NewValue.CreateInstance().GetAllModsForReference().Where(m => m.UserPlayable).Select(m => new ModButton(m)));
```

Comment:
This is kinda scary... Along with the one in `IncompatibilityDisplayingModButton`. Because the mod is used to display the tooltip.

Reply:
Are you thinking we might eventually slip up and merge an incorrect usage? Or that rulesets might do silly stuff? Second is probably a moot point though, since they could still just do it via reflection.

Reply:
The first one yes. I think we should probably use `GetAllMods()` in these cases, or at least deep-clone if they're going to actually be used, however minor the usage is (as long as it's not a pure for-reference usage).

Reply:
Hmmm... I guess? Maybe I have too much confidence in our review stage, but I can't see this particular one going wrong.

Reply:
I agree with either reverting these to `GetAllMods()` or defensive copying. I have the exact opposite feeling to the above - I have a sense of foreboding that this will _likely_ bite us somehow down the line ([case in point](https://github.com/ppy/osu/pull/14435)). But maybe it's my pessimism showing again...